### PR TITLE
[Infrastructure] Unblock installer

### PR DESCRIPTION
**Pull request by javiercn**
Split some changes from
https://github.com/dotnet/aspnetcore/pull/55339
To unblock
https://github.com/dotnet/installer/pull/19577

### CODE REVIEW

File: src/Servers/Kestrel/Core/src/KestrelServerOptions.cs
Comment (javiercn) [reply src/Servers/Kestrel/Core/src/KestrelServerOptions.cs:380]: I believe that's not the case, the previous code wasn't checking for null on the result of list certificates. `ListCertificates` doesn't return null I believe.

File: src/Identity/Specification.Tests/src/UserManagerSpecificationTests.cs
Comment (javiercn) [reply src/Identity/Specification.Tests/src/UserManagerSpecificationTests.cs:779]: This is checking `nullUserClaims == null` not `newUserClaims.Count == 0`

**Comment by javiercn** [comment]
> null-conditional operator `?[]` is a correct replacement for `FirstOrDefault`

image

**Review by TylerBrinkley — APPROVED** [review]

**Comment by tannergooding** [comment]
Commented here as well: https://github.com/dotnet/installer/pull/19577#issuecomment-2077346599, but the simpler fix here is to put `CA1826.severity` (https://github.com/dotnet/aspnetcore/blob/08e84a2fdc48d49d5a322b29a13f5b4385705639/.editorconfig#L149) back to its default, which is `suggestion`.

The linked comment has some context about the intent of the analyzer and why it can potentially be viewed as deprecated moving forward.

File: src/Identity/Specification.Tests/src/UserManagerSpecificationTests.cs
Comment (tannergooding) [inline src/Identity/Specification.Tests/src/UserManagerSpecificationTests.cs:649]: You should be able to do `logins[^1]`, notably.

Comment (javiercn) [reply src/Identity/Specification.Tests/src/UserManagerSpecificationTests.cs:649]: I think this was the code fix that did the change. I used apply fix to all occurrences in the document

**Comment by javiercn** [comment]
I've created a separate PR https://github.com/dotnet/aspnetcore/pull/55361 to change the rule to suggestion

**Comment by javiercn** [comment]
> Does this PR need to also handle
> 
> > error MSB4018: (NETCORE_ENGINEERING_TELEMETRY=Build) The "RepoTasks.RemoveSharedFrameworkDependencies" task failed unexpectedly.
> > System.MissingMethodException: Method not found: 'Boolean System.Collections.Generic.ISet`1.Contains(!0)'.    at RepoTasks.RemoveSharedFrameworkDependencies.FilterDependencies(String targetPath, ISet`1 dependencyToRemove)
> > at RepoTasks.RemoveSharedFrameworkDependencies.Execute() in /_/src/aspnetcore/eng/tools/RepoTasks/RemoveSharedFrameworkDependencies.cs:line 38
> > at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
> > at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
> 
> Which is also occurring in the blocked installer build: https://github.com/dotnet/installer/pull/19577/checks?check_run_id=24250325560
> 
> From the other PR, it sounds like this is due to project not use the shared framework workaround and so its compiling against an out of date BCL?

I think you are right, we also need a change with the build infrastructure change. I'm not familiar with that area so I think we'll need @wtgodbe for that. I'm not sure if we can separate that change into its own change that runs with the current SDK so that we can merge it and make things flow, get a new installer and then update other PRs with it.

**Comment by wtgodbe** [comment]
Here's a hacky workaround to fix the RepoTasks issue: https://github.com/dotnet/aspnetcore/pull/55339/files#diff-c86671b0d4350666f7a867a36dd59de3723a6c8288eed724bc6dc2dac3a37990L67

The runtime dependency update is blocked by some test failures and other issues, so we should just port the above to here to unblock installer

**Comment by javiercn** [comment]
@wtgodbe I've applied your suggestion here

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**
